plugins/uefi-dbx/fu-uefi-dbx-file.c

```diff
@@ -180,6 +180,68 @@ fu_uefi_dbx_file_get_checksums (FuUefiDbxFile *self)
 	return self->checksums;
 }
 
+static gboolean
+fu_uefi_dbx_file_validate_esp_volume (FuUefiDbxFile *self, FuVolume *esp, GError **error)
+{
+	g_autofree gchar *esp_path = NULL;
+	g_autoptr(GPtrArray) files = NULL;
+
+	/* get list of files contained in the ESP */
+	esp_path = fu_volume_get_mount_point (esp);
+	if (esp_path == NULL)
+		return TRUE;
+	files = fu_common_get_files_recursive (esp_path, error);
+	if (files == NULL)
+		return FALSE;
+
+	/* verify each file does not exist in the ESP */
+	for (guint i = 0; i < files->len; i++) {
+		const gchar *fn = g_ptr_array_index (files, i);
+		g_autofree gchar *checksum = NULL;
+		g_autoptr(GError) error_local = NULL;
+
+		/* get checksum of file */
+		checksum = fu_uefi_dbx_get_authenticode_hash (fn, &error_local);
+		if (checksum == NULL) {
+			g_debug ("failed to get checksum for %s: %s", fn, error_local->message);
+			continue;
```

Comment:
currently it looks like if pesign wasn't around valid files will get skipped

Reply:
This was intentional; if we can do the fallback to another tool as well maybe we can make this an error.